### PR TITLE
Decorator Creation - School Library Project

### PR DESCRIPTION
**Pull request by ouasamine**
## Overview

In this part, I created decorators classes to trim and capitalize person's names

## Workflow
- Create Nameable and BaseDecorator classes
- Link the Person class to Nameable class
- Create CapitalizeDecorator and TrimmerDecorator

### CODE REVIEW

File: decorators.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (Nemwel-Boniface) [inline decorators.rb:16]: - [ ] Great job working on the Base decorator. You are only left with creating the getter and setter for ```@nameable```. ```attr_accessor :nameable```

File: person.rb [hunk 2]
```diff
@@ -23,3 +30,10 @@ def is_of_age?
   end
   # rubocop:enable Naming/PredicateName
 end
+
+person = Person.new(22, 'maximilianus')
+p person.correct_name
+capitalized_person = CapitalizeDecorator.new(person)
+p capitalized_person.correct_name
+capitalized_trimmed_person = TrimmerDecorator.new(capitalizedPerson)
```
Comment (Nemwel-Boniface) [inline person.rb:38]: - [ ] you seem to be calling the wrong variable. There is no ```capitalizedPerson``` but it is ```person```. This is it ```capitalized_trimmed_person = TrimmerDecorator.new(person)```

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @ouasamine       :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorators.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+attr_accessor :nameable
```
Comment (danmainah) [inline decorators.rb:18]: - [ ] @ouasamine great work adding changes requested however `attr_accessor :nameable` should be added inside the BaseDecorator before the initialize method.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @ouasamine , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Gitflow was correctly used
✔️  Good design


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] [optional] It would be nice if you added the Key Features and Future Features in your readme.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @ouasamine 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Descriptive pull request 
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**